Fix static asset pipeline tests in Django 1.9.

doctoryes opened the pull request:
https://openedx.atlassian.net/browse/PLAT-1872

Continuation of work on this PR: https://github.com/edx/edx-platform/pull/17048

Ensure that node requirements are always installed before running the static asset update. Without doing this prereq install, the static asset updating fails when looking for particular JS files post-update.

doctoryes commented:
jenkins run django19 python

doctoryes commented:
jenkins run django19 python

doctoryes commented:
jenkins run django19 python

doctoryes commented:
jenkins run django19 python

doctoryes commented:
Please note that all Python tests are now passing under Django 1.9 😺

jmbowman commented on common/djangoapps/pipeline_mako/tests/test_render.py:
```diff
@@ -49,6 +50,16 @@ def setUpClass(cls):
+        if prereq_install_val_orig is None:
+            del os.environ['NO_PREREQ_INSTALL']
+        else:
+            os.environ['NO_PREREQ_INSTALL'] = prereq_install_val_orig
```
Should the environment reset be handled via `addCleanup()` so it gets run even if the test fails or hits an error?

doctoryes replied:
I'll try putting the env reset in a `tearDownClass` since it's overridden in `setUpClass`.

doctoryes commented:
jenkins run django19 python

doctoryes commented:
jenkins run python

doctoryes commented:
jenkins run python

pwnage101 commented:
Since the cleanup logic was moved, update_assets paver task will possibly run with NO_PREREQ_INSTALL set, which will invoke 'pavelib.prereqs.install_node_prereqs' again.  Is that okay, since the task is idempotent, or perhaps this adds some extra time by installing things twice?

doctoryes commented:
@pwnage101 That setup/teardown is placed in `setUpClass/tearDownClass` - so it will only run once instead of for each test in the class.

doctoryes commented:
jenkins run python

jmbowman commented:
jenkins run django19 bokchoy

jmbowman commented:
jenkins run django19 lettuce

doctoryes commented:
jenkins run django19 python

doctoryes commented:
jenkins run django19 bokchoy

doctoryes commented:
jenkins run django19 lettuce

doctoryes commented:
jenkins run django19 python

doctoryes commented:
Please re-review. @jmbowman I removed the `tearDownClass` change - but I added a try/except/finally block to set the env var back to the correct value.

doctoryes commented:
jenkins run django19 python

jmbowman commented:
Fine with me.

doctoryes commented:
jenkins run lettuce
